### PR TITLE
chore: Cache SSH connections

### PR DESCRIPTION
**Pull request by kylecarbs**
Speedy speedy speed...

Works for Linux and Darwin systems, [but not Windows](https://github.com/microsoft/vscode-remote-release/issues/629#issuecomment-503650294).

https://user-images.githubusercontent.com/7122116/124811861-8e0f8600-df28-11eb-8925-51a301e5ff54.mp4

### CODE REVIEW

File: internal/cmd/configssh.go [hunk 1]
```diff
@@ -208,7 +209,7 @@ func makeNewConfigs(binPath, userName string, workspaces []coderutil.WorkspaceWi
 
 func makeSSHConfig(binPath, host, userName, workspaceName, privateKeyFilepath string, tunnel bool) string {
 	if tunnel {
-		return fmt.Sprintf(
+		host := fmt.Sprintf(
```
Comment (jawnsy) [inline internal/cmd/configssh.go:212]: Is it possible/straightforward to add an integration test for this?

**Review by jawnsy — APPROVED** [review]
Nice work! Does this also multiplex connections over a single TCP stream, or was that a separate feature?

File: internal/cmd/configssh.go (hunk 1 above)
Comment (kylecarbs) [reply internal/cmd/configssh.go:212]: Good idea.... will do!

**Comment by kylecarbs** [comment]
It multiplexes over a single TCP stream!

**Review by f0ssel — APPROVED** [review]

**Pull request merged**